lib/studies/reply/adaptive_nav.dart

```diff
@@ -791,6 +732,99 @@ class _MobileNavState extends State<_MobileNav> with TickerProviderStateMixin {
   }
 }
 
+class _AnimatedBottomAppBar extends StatelessWidget {
+  const _AnimatedBottomAppBar({
+    this.bottomAppBarController,
+    this.bottomAppBarCurve,
+    this.bottomDrawerVisible,
+    this.drawerController,
+    this.dropArrowCurve,
+    this.navigationDestinations,
+    this.selectedIndex,
+    this.toggleBottomDrawerVisibility,
+  });
+
+  final AnimationController bottomAppBarController;
+  final Animation<double> bottomAppBarCurve;
+  final bool bottomDrawerVisible;
+  final AnimationController drawerController;
+  final Animation<double> dropArrowCurve;
+  final List<_Destination> navigationDestinations;
+  final int selectedIndex;
+  final VoidCallback toggleBottomDrawerVisibility;
+
+  @override
+  Widget build(BuildContext context) {
+    return Selector<EmailStore, bool>(
+      selector: (context, emailStore) => emailStore.onMailView,
+      builder: (context, onMailView, child) {
+        bottomAppBarController.forward();
```

Comment:
I don't think bottomAppBarController needs to be passed down. You can do this in the parent's lifecycle functions

Reply:
I tried putting it in `didUpdateWidget` and `didChangeDependencies`, but those aren't called when we transition to the `MailViewPage`.